HIF.md

```diff
@@ -0,0 +1,810 @@
+|`ENOMEM`              |`0x0005`| Not enough memory available.                                |
+|`EINVAL`              |`0x1000`| Invalid parameter (generic).                                |
+|`EINVAL_HOSTAPI`      |`0x1001`| Incompatible host API version.                              |
+|`EINVAL_PHY`          |`0x1002`| Invalid PHY selection.                                      |
+|`EINVAL_TXPOW`        |`0x1003`| Invalid TX power.                                           |
+|`EINVAL_REG`          |`0x1004`| Invalid regulation code ([`SET_RADIO_REGULATION`][rf-reg]). |
+|`EINVAL_FHSS`         |`0x1005`| Invalid frequency hopping configuration (generic).          |
+|`EINVAL_FHSS_TYPE`    |`0x1006`| Invalid FHSS type ([`REQ_DATA_TX`][tx-req]).                |
+|`EINVAL_CHAN_MASK`    |`0x1007`| Invalid channel mask.                                       |
+|`EINVAL_CHAN_FUNC`    |`0x1008`| Invalid channel function.                                   |
+|`EINVAL_ASYNC_TXLEN`  |`0x1009`| Invalid asynchronous transmission maximum duration ([`SET_FHSS_ASYNC`][async]).|
+|`EINVAL_HANDLE`       |`0x100a`| Invalid packet handle.                                      |
+|`EINVAL_KEY_INDEX`    |`0x100b`| Invalid key index.                                          |
+|`EINVAL_FRAME_LEN`    |`0x100c`| Invalid IEEE 802.15.4 frame length.                         |
+|`EINVAL_FRAME_VERSION`|`0x100d`| Invalid IEEE 802.15.4 frame version.                        |
+|`EINVAL_FRAME_TYPE`   |`0x100c`| Invalid IEEE 802.15.4 frame type.                           |
+|`EINVAL_ADDR_MODE`    |`0x100e`| Invalid IEEE 802.15.4 address mode.                         |
+|`EINVAL_SCF`          |`0x100f`| Invalid IEEE 802.15.4 security control field.               |
+|`EINVAL_FRAME`        |`0x1010`| Invalid IEEE 802.15.4 frame (generic).                      |
+|`EINVAL_CHAN_FIXED`   |`0x1011`| Invalid fixed channel.                                      |
+|`ENOTSUP`             |`0x2000`| Unsupported feature (generic).                              |
+|`ENOTSUP_FHSS_DEFAULT`|`0x2001`| Unsupported configuration mode for selected FHSS type ([`REQ_DATA_TX`][tx-req]).|
+
+### `0x06 SET_HOST_API`
+
+Inform the RCP about the host API version. If not send, the RCP will assume
+`2.0.0`. This command should be sent after `IND_RESET`, before any other
+command.
+
+ - `uint32_t api_version`  
+    Must be at least `0x02000000`.
+
+## Send and receive data
+
+Only a subset of the IEEE 802.15.4 frame formats are supported for the needs of
+Wi-SUN:
+
+  - Frame type must be `1` (data) or `2` (ack, [`CNF_DATA_TX`][tx-cnf] only).
+  - Frame version must be `2`.
+  - Source address mode must be `3`.
+  - Destination address mode must be either `0` or `3`.
+  - See ["Security"][sec] for limits on the auxiliary security header.
+
+[cca]:    #channel-access-and-retries
+[tx-req]: #0x10-req_data_tx
+[tx-cnf]: #0x12-cnf_data_tx
+[rx]:     #0x13-ind_data_rx
+
+### Channel Access and Retries
```

Comment:
I think this is necessary to explain `cca_failures` and `tx_failures` in `CNF_DATA_TX`.
Also some users might be interested in the CSMA-CA parameters in use, although I am aware that documenting this may raise unwanted questions from customers.